MovieApp/MovieApp/Model/Managers/MovieManager.swift

```diff
@@ -72,4 +179,95 @@ class MovieManager {
+    func getDetails(movieID: Int?) {
+        reachability.whenReachable = { _ in
+           self.getMovieDetails(id: movieID)
+        }
+        
+        reachability.whenUnreachable = { _ in
+            if !self.getMovieDetailsRealm(id: movieID) {
+                NotificationCenter.default.post(name: Notification.Name(Constants.Network.updateNetworkStatus), object: nil, userInfo: [Constants.Network.updateNetworkStatus : Constants.Network.statusOffline])
+            }
+        }
+        self.configureNetwork()
+    }
+    
+    func getMovieDetails(id: Int?) {
+        self.getMovieDetailsAPI(id: id) { movie in
+            DispatchQueue.main.async {
+                self.realmRepository.addMovieDetails(movie: movie)
+                self.bannerOfflineMovieID = movie.id
+                self.getMovieRating(completion: self.saveMovieRating)
+            }
+            self.delegate?.onBannerLoaded()
+        }
+    }
+    
+    func getMovieDetailsAPI(id: Int?, completion: @escaping (Movie) -> Void) {
+        self.repository.getMovieDetails(id: id) { result in
+            switch result {
+            case .success(let movie):
+                self.movie = movie
+                completion(movie)
+                break
+            case .failure(let error):
+                self.errorDelegate?.showAlertMessage(title: Constants.General.errorTitle, message: error.rawValue)
+            }
+        }
+    }
+    
+    func getMovieDetailsRealm(id: Int?) -> Bool {
+        guard let movieRealm = self.realmRepository.getMovieDetails(id: id) else {
+            return false
+        }
+        self.movie = Movie(movie: movieRealm)
+        self.delegate?.onBannerLoaded()
+        return true
+    }
+    
+    func getMovieRating(completion: @escaping (Movie) -> Void) {
+        self.repository.getMovieRating(id: self.movie?.imdbID) { result in
+            switch result {
+            case .success(let movie):
+                guard let response = movie.response, response == "True", let rating = movie.rating else {
+                    self.errorDelegate?.showAlertMessage(title: Constants.General.errorTitle, message: CustomError.notFoundError.rawValue)
+                    return
+                }
+                self.movie?.rating = Double(rating)
+                completion(self.movie!)
+                break
+            case .failure(let error):
+                self.errorDelegate?.showAlertMessage(title: Constants.General.errorTitle, message: error.rawValue)
+            }
+        }
+    }
+    
+    func saveMovieRating(movie: Movie) {
+        DispatchQueue.main.async {
+            self.realmRepository.addMovieDetails(movie: movie)
+        }
+        if let rating = movie.rating {
+            self.delegate?.didSetRating(rating)
+            self.delegate?.onBannerLoaded()
+        }
+    }
+    
+    func getCast(at index: Int) -> Cast {
+        return self.cast?[index] ?? Cast(id: 9, name: "", profilePath: "", character: "")
```

Comment:
It's better if you return nil here instead of mocked Cast, and use guard let or similar in VC to check for this case